dcpy/models/base.py

```diff
@@ -131,3 +132,119 @@ def from_yaml(cls, yaml_str: str, *, template_vars=None) -> typing.Self:
     def from_path(cls, path: Path, *, template_vars=None) -> typing.Self:
         with open(path, "r", encoding="utf-8") as raw:
             return cls.from_yaml(raw.read(), template_vars=template_vars)
+
+
+class IndentedPrint(BaseModel, arbitrary_types_allowed=True):
```

Comment:
Should this just be a function? The parameters to reminder me a lot of the `tabulate` api. Upside of a function is that any model can use it without inheriting from it.